app/screens/thread/index.js

```diff
@@ -4,29 +4,36 @@
 import {bindActionCreators} from 'redux';
 import {connect} from 'react-redux';
 
-import {getTheme} from '@mm-redux/selectors/entities/preferences';
-
 import {selectPost} from '@mm-redux/actions/posts';
+import {setThreadFollow, updateThreadRead} from '@mm-redux/actions/threads';
+import {getCurrentUserId} from '@mm-redux/selectors/entities/common';
 import {getChannel, getMyCurrentChannelMembership} from '@mm-redux/selectors/entities/channels';
 import {makeGetPostIdsForThread} from '@mm-redux/selectors/entities/posts';
+import {getTheme, isCollapsedThreadsEnabled} from '@mm-redux/selectors/entities/preferences';
+import {getCurrentTeamId} from '@mm-redux/selectors/entities/teams';
+import {getThread} from '@mm-redux/selectors/entities/threads';
 
 import Thread from './thread';
 
 function makeMapStateToProps() {
     const getPostIdsForThread = makeGetPostIdsForThread();
     return function mapStateToProps(state, ownProps) {
         const channel = getChannel(state, ownProps.channelId);
-
+        const collapsedThreadsEnabled = isCollapsedThreadsEnabled(state);
         return {
             channelId: ownProps.channelId,
+            channelIsArchived: channel ? channel.delete_at !== 0 : false,
             channelType: channel ? channel.type : '',
+            collapsedThreadsEnabled,
             displayName: channel ? channel.display_name : '',
             myMember: getMyCurrentChannelMembership(state),
-            rootId: ownProps.rootId,
             postIds: getPostIdsForThread(state, ownProps.rootId),
+            rootId: ownProps.rootId,
+            teamId: getCurrentTeamId(state),
             theme: getTheme(state),
-            channelIsArchived: channel ? channel.delete_at !== 0 : false,
+            thread: getThread(state, ownProps.rootId, true),
             threadLoadingStatus: state.requests.posts.getPostThread,
+            userId: getCurrentUserId(state),
```

Comment:
perhaps make this clearer as in 
```suggestion
            currentUserId: getCurrentUserId(state),
```